### PR TITLE
Unify Regions with RegionVids in UnificationTable

### PR DESCRIPTION
**Pull request by jackh726**
A few test output changes; might be able to revert those but figured I would open this for perf and comments.

r? @nikomatsakis

### CODE REVIEW

**Review by nikomatsakis — COMMENTED** [review]
I left some thoughts on Zulip.

File: compiler/rustc_infer/src/infer/region_constraints/mod.rs [hunk 1]
```diff
@@ -617,8 +623,25 @@ impl<'tcx> RegionConstraintCollector<'_, 'tcx> {
         }
     }
 
-    pub fn opportunistic_resolve_var(&mut self, rid: RegionVid) -> ty::RegionVid {
-        self.unification_table().probe_value(rid).min_vid
+    pub fn opportunistic_resolve_var(&mut self, rid: ty::RegionVid) -> ty::RegionVid {
+        self.unification_table().find(rid).vid
+    }
+
+    pub fn opportunistic_resolve_region(
```
Comment (nikomatsakis) [inline compiler/rustc_infer/src/infer/region_constraints/mod.rs:630]: This function is dead code in this PR, right?

File: compiler/rustc_infer/src/infer/region_constraints/mod.rs [hunk 2]
```diff
@@ -617,8 +623,25 @@ impl<'tcx> RegionConstraintCollector<'_, 'tcx> {
         }
     }
 
-    pub fn opportunistic_resolve_var(&mut self, rid: RegionVid) -> ty::RegionVid {
-        self.unification_table().probe_value(rid).min_vid
+    pub fn opportunistic_resolve_var(&mut self, rid: ty::RegionVid) -> ty::RegionVid {
```
Comment (nikomatsakis) [inline compiler/rustc_infer/src/infer/region_constraints/mod.rs:626]: We should comment on the distinction i behavior between these two functions!

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: compiler/rustc_infer/src/infer/region_constraints/mod.rs (hunk 1 above)
Comment (jackh726) [reply compiler/rustc_infer/src/infer/region_constraints/mod.rs:630]: Yes, would you rather me remove it?

File: compiler/rustc_infer/src/infer/region_constraints/mod.rs (hunk 2 above)
Comment (jackh726) [reply compiler/rustc_infer/src/infer/region_constraints/mod.rs:627]: Added

**Review by nikomatsakis — APPROVED** [review]

File: compiler/rustc_infer/src/infer/region_constraints/mod.rs (hunk 1 above)
Comment (nikomatsakis) [reply compiler/rustc_infer/src/infer/region_constraints/mod.rs:634]: Not necessary, just checking.

**Pull request merged**